### DISCUSSION TITLE
Is there a way to get the connection status?

### DISCUSSION THREAD

**Discussion by Unit111** [miguelgrinberg/python-socketio#796] [Q&A]
```
    def comment_added(data, context, sio):
        print(f"RECEIVING DATA FROM SERVER EMIT {data}")
        context.last_response = data
        sio.disconnect()

    sio.connect(context.url, context.token)
    time.sleep(3)
    sio.emit("[Conversations] Subscribe to Event for Conversations",
             data=("[Conversations] Conversation Comment Added", conv))
    make_post(context, body, endpoint)
    sio.on('[Conversations] Conversation Comment Added',
           handler=lambda data, context=context, sio=sio: comment_added(data, context, sio))
    sio.wait()
    print(context.last_response)
```

I know this doesn't look good but I am writing acceptance tests so I need to make this workflow synchronous. What I need is to call the emil after the connection is established. Currently this is done with the sleep but there is no chance I'm pushing this code.
I can do this in a couple of ways:
1. Similar to the way I'm handling the '[Conversations] Conversation Comment Added' event can I register the event handler for the 'connect' event as a method? The documentation states that this event is reserved so I guess not.
2. If there is a way to get the connection status I will be able to remove the time.sleep and call emil only after the connection is established.

**Comment by miguelgrinberg** [comment]
What version of python-socketio are you using? In recent versions, when the `connect()` call returns you can emit right away.

**Comment by Unit111** [comment]
python-socketio[client]==4.0.0
python-engineio==3.2.0
These are the newest ones compatible with the product I am testing.

**Reply by miguelgrinberg** [reply]
For such an old release the only thing you can do is use an `Event` object instead of the sleep. You can signal the event in the connect handler. This is actually more or less what newer versions do to delay the return of the connect function until when it is safe to emit.
